internal/controller/workspace_controller.go

```diff
@@ -112,14 +112,14 @@ func (r *WorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	}
 
 	// Ensure template label is set if workspace uses a template
-	if workspace.Spec.TemplateRef != nil && *workspace.Spec.TemplateRef != "" {
+	if workspace.Spec.TemplateRef != nil && workspace.Spec.TemplateRef.Name != "" {
```

Comment:
if the `TemplateRef` was removed by the update (i.e. w/ template > no template), this block wouldn't trigger whereas the label ought to be removed, would it?

My 2c: this doesn't belong to the controller, move to the webhook please. Add the label at the same time as we add the `created-by` / `last-updated-by` annotations in the [defaulting webhook](https://github.com/jupyter-infra/jupyter-k8s/blob/296de4170ccbb7fedf8a5974b06fd86e5efc223b/internal/webhook/v1alpha1/workspace_webhook.go#L147)